pkg/agent/qrm-plugins/cpu/vcipolicy/policy.go

```diff
@@ -0,0 +1,392 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vcipolicy
+
+import (
+	"context"
+	"fmt"
+	"github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/cpu/vcipolicy/vci"
+	utilkubeconfig "github.com/kubewharf/katalyst-core/pkg/util/kubelet/config"
+	"k8s.io/klog/v2"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/kubewharf/katalyst-api/pkg/plugins/skeleton"
+	"github.com/kubewharf/katalyst-core/cmd/katalyst-agent/app/agent"
+	cpuconsts "github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/cpu/consts"
+	"github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/cpu/vcipolicy/state"
+	"github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/util"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/config/agent/dynamic/crd"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+	"github.com/kubewharf/katalyst-core/pkg/util/machine"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	pluginapi "k8s.io/kubelet/pkg/apis/resourceplugin/v1alpha1"
+)
+
+const (
+	// cpuPluginStateFileName is the name of cpu plugin state file.
+	cpuPluginStateFileName = "cpu_plugin_state"
+)
+
+// NativePolicy is a policy compatible with Kubernetes native semantics and is used in topology-aware scheduling scenarios.
+type VCIStaticPolicy struct {
```

Comment:
Is there a more generic name than "VCI" policy?

Reply:
Maybe some constants containing 'vci.volcengine.com' can be passed through args to avoid the appearance of these non-public naming in the code ?

Reply:
Is this policy suitable for open source?